app/assets/javascripts/host_power.js

```diff
@@ -0,0 +1,64 @@
+(function(){
+
+  $(document).ready(function () {
+    powerStatus();
+  });
+
+  function hostIds() {
+    var res = [];
+
+    $('#hosts-table tbody tr').each( function (index, item) {
+      res.push($(item).data("host-id"));
+    });
```

Comment:
aha, I found the reason. `.map()` returns a jQuery object, not an array. To get the array you need to call `.get()` on it, like so:
```javascript
return $('#hosts-table tbody tr').map(function(){return $(this).data("host-id")}).get();
```

but I had another thought - since you use this function to get the rows to process, and except for the ajax data you don't actually need to go over the id list but would be able to just go over the rows, perhaps you could just return the rows and move the mapping to the ajax function?